docs/docsite/rst/user_guide/playbooks_filters.rst

```diff
@@ -1522,14 +1522,19 @@ The ``regex_search`` filter returns an empty string if it cannot find a match::
     {{ 'ansible' | regex_search('foobar') }}
     # => ''
 
-Note that due to historic behavior and custom re-implementation of some of the Jinja internals in Ansible there is an exception to the behavior. When used in a Jinja expression (for example in conjunction with operators, other filters, and so on) the return value differs, in those situations the return value is ``none``. See the two examples below::
+.. note::
+
+
+  The ``regex_search`` filter returns ``none`` when used in a Jinja expression (for example in conjunction with operators, other filters, and so on). See the two examples below.
+
+  .. code-block:: Jinja
 
     {{ 'ansible' | regex_search('foobar') == '' }}
     # => False
     {{ 'ansible' | regex_search('foobar') == none }}
     # => True
 
-When ``jinja2_native`` setting is enabled, the ``regex_search`` filter always returns ``none`` if it cannot find a match.
+  This is due to historic behavior and the custom re-implementation of some of the Jinja internals in Ansible. Enable the ``jinja2_native`` setting if you want the ``regex_search`` filter to  always return ``none`` if it cannot find a match.
```

Comment:
```suggestion
  This is due to historic behavior and the custom re-implementation of some of the Jinja internals in Ansible. Enable the ``jinja2_native`` setting if you want the ``regex_search`` filter to  always return ``None`` if it cannot find a match.
```

Capitalization here too.

Reply:
We probably need a link from this, to a more comprehensive explanation about this behavior in another part of the docs.  Maybe in an FAQ.

I'll see if I can detail it a bit here, but feel free to reach out to me directly for any needed clarification:

Non-native jinja2 is the historical integration with jinja2.  What this means is that jinja2 has historically only been able to return strings.  Which means that ansible has needed a way to convert python string representations of python objects, back into a python object.  We do this using an internal ansible feature called `safe_eval`.  We only send strings that look like certain types of python objects through this function. Ultimately, `None` would be converted to the string `"None"` which isn't particularly useful in non-native jinja2.  As a result a single templating action such as:

```
{{ 'ansible' | regex_search('foobar') }}
```

Cannot produce a Python `None`, so we have historically converted it to `""` (empty string).  The use of "result a single templating action" is important here, as it means the final result of the templating action.

When doing:

```
{{ 'ansible' | regex_search('foobar') is none }}
```

The output of `regex_search` is not the final result, it's an intermediate result that is then compared to the jinja2 `none` test.

The native jinja2 functionality actually allows us to return full python objects, that are always represented as python objects everywhere, and as such a "result a single templating action" can result in the Python `None`.

I hope this helps clarify this.